mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs

```diff
@@ -452,10 +455,14 @@ fn global_nft_with_random_privkey(
         path_to_address: Default::default(),
         gap_limit: None,
     };
-
-    ensure_p2p_context(&MM_CTX1);
+    let ctx = MmCtxBuilder::new()
+        .with_conf(json!({"use_trading_proto_v2": true}))
+        .into_mm_arc();
+    ctx.init_metrics().unwrap();
```

Comment:
are metrics really needed in unit test?

Reply:
not really, the issue is not in metrics.

I started to have problems after dev merge with p2p updates
https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10470145546/job/28997224059
```
---- docker_tests::eth_docker_tests::send_and_refund_erc1155_maker_payment_secret stdout ----
thread 'docker_tests::eth_docker_tests::send_and_refund_erc1155_maker_payment_secret' panicked at 'called `Option::unwrap()` on a `None` value', mm2src/mm2_net/src/p2p.rs:37:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- docker_tests::eth_docker_tests::send_and_refund_erc1155_maker_payment_timelock stdout ----
thread 'docker_tests::eth_docker_tests::send_and_refund_erc1155_maker_payment_timelock' panicked at 'called `Result::unwrap()` on an `Err` value: PoisonError { .. }', mm2src/mm2_net/src/p2p.rs:35:14

---- docker_tests::eth_docker_tests::send_and_refund_erc721_maker_payment_secret stdout ----
thread 'docker_tests::eth_docker_tests::send_and_refund_erc721_maker_payment_secret' panicked at 'called `Result::unwrap()` on an `Err` value: PoisonError { .. }', mm2src/mm2_net/src/p2p.rs:35:14

----  etc
```
in tests Im using `eth_coin_from_conf_and_request_v2` fnc which has `build_web3_instances`, which after update has `let p2p_ctx = P2PContext::fetch_from_mm_arc(ctx);` MM CTX for docker tests return none for p2pctx thats why I had issues above.

So Im trying to init p2p ctx and also for this I need `CryptoCtx::from_ctx(ctx)`.  after providing p2p ctx and cryptro ctx tests get stuck endlesslyhttps://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/10472900751/job/29003513158

so the problem above is what im trying to fix.